tools/reckless

```diff
@@ -0,0 +1,684 @@
+#!/usr/bin/env python3
+
+from subprocess import Popen, PIPE
+import sys
+import json
+import os
+import argparse
+from pathlib import Path
+import shutil
+import tempfile
+import requests
+
+
+repos = ['https://github.com/lightningd/plugins']
+
+
+def py_entry_guesses(name):
+    return [name, f'{name}.py', '__init__.py']
+
+
+def unsupported_entry(name):
+    return [f'{name}.go', f'{name}.sh']
+
+
+class InstInfo:
+    def __init__(self, name, url, git_url):
+        self.name = name
+        self.repo = url         # Used for 'git clone'
+        self.git_url = git_url  # API access for github repos
+        self.entry = None
+        self.deps = None
+        self.subdir = None
+        self.commit = None
+
+    def __repr__(self):
+        return f'<InstInfo object>\n'\
+               f'name: {self.name}\nrepo: {self.repo}\ngit: {self.git_url}'\
+               f'\nentry:{self.entry}\ndepency source:{self.deps}'
+
+    def get_inst_details(self):
+        """
+        Populate installation details from a github repo url.
+        Return True if all data is found.
+        """
+        r = requests.get(self.git_url)
+        if r.status_code != 200:
+            return False
+        if 'git/tree' in self.git_url:
+            tree = r.json()['tree']
+        else:
+            tree = r.json()
+        entry_guesses = py_entry_guesses(self.name)
+        for g in entry_guesses:
+            for f in tree:
+                if f['path'] == g:
+                    self.entry = g
+                    break
+            if self.entry is not None:
+                break
+        if self.entry is None:
+            for g in unsupported_entry(self.name):
+                for f in tree:
+                    if f['path'] == g:
+                        # FIXME: This should be easier to implement
+                        print(f'entrypoint {g} is not yet supported')
+                        return False
+        dependency_info = ['requirements.txt', 'pyproject.toml']
+        for d in dependency_info:
+            for f in tree:
+                if f['path'] == d:
+                    self.deps = d
+                    break
+            if self.deps is not None:
+                break
+        if not self.entry:
+            return False
+        if not self.deps:
+            return False
+        return True
+
+
+def create_dir(r, directory):
+    """Creation of a directory at path `d` with a maximum new dir depth `r`"""
+    if os.path.exists(directory):
+        return True
+    elif r <= 0:
+        return False
+    elif create_dir(r-1, os.path.split(directory)[0]):
+        os.mkdir(directory, 0o777)
+        print(f'created directory {directory}')
+        if os.path.exists(directory):
+            return True
```

Comment:
This can be achieved by `pathlib.Path(directory).mkdir(parents=True, exist_ok=True)`: https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir